### PR TITLE
Implement basic geometry functions

### PR DESCRIPTION
**Pull request by romanroibu**
Implemented some basic geometry functions

### CODE REVIEW

File: Foundation/NSGeometry.swift [hunk 1]
```diff
@@ -53,6 +69,12 @@ public struct CGSize {
     }
 }
 
+extension CGSize: Equatable { }
+
+public func ==(lhs: CGSize, rhs: CGSize) -> Bool {
+    return (lhs.width == rhs.width) && (lhs.height == lhs.height)
```
Comment (johankool) [inline Foundation/NSGeometry.swift]: `lhs.height == ls.height` is always true. You want `lhs.height == rhs.height`

File: Foundation/NSGeometry.swift [hunk 2]
```diff
@@ -40,6 +50,12 @@ public struct CGPoint {
     }
 }
 
+extension CGPoint: Equatable { }
+
+public func ==(lhs: CGPoint, rhs: CGPoint) -> Bool {
+    return (lhs.x == rhs.x) && (lhs.y == lhs.y)
```
Comment (johankool) [inline Foundation/NSGeometry.swift]: You want `lhs.y == rhs.y` here instead.

File: Foundation/NSGeometry.swift [hunk 3]
```diff
@@ -66,6 +88,12 @@ public struct CGRect {
     }
 }
 
+extension CGRect: Equatable { }
+
+public func ==(lhs: CGRect, rhs: CGRect) -> Bool {
+    return (lhs.origin == rhs.origin) && (lhs.size == lhs.size)
```
Comment (johankool) [inline Foundation/NSGeometry.swift]: You want `lhs.size == rlhs.size` here instead.

Comment (romanroibu) [inline Foundation/NSGeometry.swift]: Wow, can't believe I've missed those. Thanks!

**Comment by parkera** [comment]
Let's rebase this after incorporating @johankool 's comments so we can merge it in. Thanks!

**Comment by romanroibu** [comment]
Done, thanks!

File: Foundation/NSGeometry.swift [hunk 4]
```diff
@@ -66,6 +88,12 @@ public struct CGRect {
     }
 }
 
+extension CGRect: Equatable { }
+
+public func ==(lhs: CGRect, rhs: CGRect) -> Bool {
```
Comment (phausler) [inline Foundation/NSGeometry.swift]: This functionality on Darwin actually lives in CoreGraphics not in Foundation

**Comment by phausler** [comment]
Looks fine to me.

**Pull request merged**